Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleFrame.cpp

```diff
@@ -299,3 +299,29 @@ void _WKBundleFrameGenerateTestReport(WKBundleFrameRef frameRef, WKStringRef mes
     if (RefPtr document = coreFrame->document())
         document->reportingScope().generateTestReport(WebKit::toWTFString(message), WebKit::toWTFString(group));
 }
+
+void* _WKAccessibilityRootObject(WKBundleFrameRef frameRef)
+{
+    RefPtr frame = WebKit::toImpl(frameRef)->coreLocalFrame();
+    if (!frame)
+        return nullptr;
+
+    while (auto* parent = dynamicDowncast<WebCore::LocalFrame>(frame->tree().parent()))
```

Comment:
LocalFrame::rootFrame can be called instead of this.  We store a pointer in order to avoid loops like this.